cub/cub/device/dispatch/dispatch_transform.cuh

```diff
@@ -197,13 +208,13 @@ struct dispatch_t<StableAddress,
 
       elem_counts last_counts{};
       // Increase the number of output elements per thread until we reach the required bytes in flight.
-      static_assert(policy_t::min_items_per_thread <= policy_t::max_items_per_thread, ""); // ensures the loop below
+      _CCCL_ASSERT_HOST(policy.MinItemsPerThread() <= policy.MaxItemsPerThread(), ""); // ensures the loop below
```

Comment:
Important: It's really sad if we have to demote compile-time checks to runtime checks. I think there is a macro defined when CUB is built for `cuda.parallel`. I think we should use it to generate a runtime check for `cuda.parallel` and stay with the compile-time check for CUB.

Reply:
Hmm, I guess I could key this off of `CUB_DEFINE_RUNTIME_POLICIES`, since that is pretty much only relevant then. I am less convinced about the assert, to be quite honest, but the argument about doing something similar for the `if constexpr` change you mentioned below makes perfect sense to me.

Reply:
We can use `CCCL_C_EXPERIMENTAL` as a macro. There is precedence for it.